bots/pr/src/main/java/org/openjdk/skara/bots/pr/ApprovalCommand.java

```diff
@@ -0,0 +1,144 @@
+        writer.println(APPROVAL_MARKER);
+    }
+
+    private void disapprovalReply(PullRequest pr, PrintWriter writer) {
+        writer.println(String.format("@%s this pull request was rejected by the maintainer. "
+                + "The bot will close this pull request automatically.", pr.author().username()));
+        writer.println(DISAPPROVAL_MARKER);
+    }
+
+    @Override
+    public void handle(PullRequestBot bot, PullRequest pr, CensusInstance censusInstance, Path scratchPath,
+                       CommandInvocation command, List<Comment> allComments, PrintWriter reply, PullRequestWorkItem workItem) {
+        if (!workItem.isUpdateChange()) {
+            reply.println("this repository or the target branch of this pull request have not been configured to use the `approval` command.");
+            return;
+        }
+
+        var commandUser = censusInstance.namespace().get(command.user().id());
+        if (!workItem.isMaintainer(commandUser)) {
+            reply.println("only the repository maintainers are allowed to use the `approval` command.");
+            return;
+        }
+
+        var arg = command.args().trim().toLowerCase();
+        if (!arg.isEmpty() && !(arg.equals("yes") || arg.equals("y") || arg.equals("no") || arg.equals("n"))) {
+            showHelp(reply);
+            return;
+        }
+
+        if (arg.equals("no") || arg.equals("n")) {
+            for (var vcsIssue : workItem.issues(false, false)) {
+                var issueOpt = bot.issueProject().issue(vcsIssue.shortId());
+                issueOpt.ifPresent(issue -> {
+                    if (!issue.labelNames().contains(workItem.requestLabelName())) {
+                        // The maintainers may disapprove the PR before it is ready.
+                        // The bot should add the fix request label firstly to avoid the strange
+                        // middle state which has disapproval label but has no fix request label.
+                        issue.addLabel(workItem.requestLabelName());
+                    }
+                    if (issue.labelNames().contains(workItem.approvalLabelName())) {
+                        // If the maintainers have approved the PR before,
+                        // the bot should remove the approval label at first.
+                        issue.removeLabel(workItem.approvalLabelName());
+                    }
+                    if (!issue.labelNames().contains(workItem.disapprovalLabelName())) {
+                        issue.addLabel(workItem.disapprovalLabelName());
+                    }
+                });
+            }
+            if (pr.labelNames().contains("approval")) {
+                pr.removeLabel("approval");
+            }
+            disapprovalReply(pr, reply);
+            pr.setState(Issue.State.CLOSED);
+            return;
+        }
+
+        pr.setState(Issue.State.OPEN);
```

Comment:
The pull request would be automatically closed by the bot if the maintainer had rejected the pull request before. And at second thought, the maintainer approves the pull request, we need to open it to confirm the CheckWorkItem can be re-run in this pull request.

But now I find a bug about opening/closing the pull request automatically. When the maintainer has rejected the pull request, the pull request was closed by the bot. But if the author of the pull request re-open the pull request by using the `open` command (the author may want to change the pull request), the bot will re-close the pull request again because the disapproval label still exists. This bug seems not easy to solve now, so I suggest that the bot doesn't set the pull request state automatically. What is your opinion.